src/upstream.c

```diff
@@ -343,7 +345,7 @@ void *processUpstreamMessage()
 								Expecting dest format as mac:xxxxxxxxxxxx/parodus/cloud-status
 								Parse dest field and check destService is "parodus" and destApplication is "cloud-status"
 							*/
-							if(destService != NULL && destApplication != NULL && strcmp(destService,"parodus")== 0 && strcmp(destApplication,"cloud-status")== 0)
+							if(macId != NULL && destService != NULL && destApplication != NULL && strcmp(destService,"parodus")== 0 && strcmp(destApplication,"cloud-status")== 0)
```

Comment:
macId != NULL should be the main outer if check. All these if, else if and else should be inside it.
In the else of this mac not null check we should log mac null error.